cpp/examples/nested_types/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+cmake_minimum_required(VERSION 3.26.4)
+
+project(
+  nested_types
+  VERSION 0.0.1
+  LANGUAGES CXX CUDA
+)
+
+include(../fetch_dependencies.cmake)
+
+# Configure your project here
+add_executable(deduplication deduplication.cpp)
+target_link_libraries(deduplication PRIVATE cudf::cudf)
+target_compile_features(deduplication PRIVATE cxx_std_17)
```

Comment:
`target_compile_features` is the newer option. As for why to prefer it, in this case I don't think it matters too much although there are subtle differences in how it interacts with other CMake variables like `CMAKE_CXX_STANDARD`. It would be more helpful if you wanted it to not be `PRIVATE` but instead `PUBLIC`/`INTERFACE` because compile features will propagate to consuming targets while properties won't.

Reply:
The `CXX_STANDARD` property is a request for a compiler level, but can decay to the highest supported level. You need to also set `CXX_STANDARD_REQUIRED` to stop the decaying.

`target_compile_features(deduplication PRIVATE cxx_std_17)` is a hard requirement and errors out instead of decaying.

Reply:
Thanks, both of you.

Reply:
I struggle with knowing what "the newer/better option" is most of the time. 😕